aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -405,12 +406,13 @@ std::tuple<Tensor, Tensor, Tensor> batch_norm_cuda_template(const Tensor& input_
   int64_t n_input = input_.size(1);
   Tensor save_mean_;
   Tensor save_invstd_;
-  auto input_cont = input_.reshape({input_.size(0), input_.size(1), -1}); // internally we merge the feature dimensions
-  auto output_cont = at::empty_like(input_cont);
+  auto input_reshaped = input_.reshape({input_.size(0), input_.size(1), -1}); // internally we merge the feature dimensions
+  auto output_reshaped = at::empty_like(input_reshaped);
 
-  auto bs = input_cont.size(0);
-  auto features = input_cont.size(2);
-  auto input = input_cont.packed_accessor<scalar_t, 3, RestrictPtrTraits>();
+  AT_CHECK(cuda::detail::canUse32BitIndexMath(input_reshaped), "Input is too large for batch_norm");
```

Comment:
hmm can you instead template the whole thing and use `int32_t` if `canUse32BitIndexMath` and `int64_t` otherwise?